### PR TITLE
[Core][LUA] Refactor Moon Cycle Handling

### PR DESCRIPTION
**Pull request by Rosienaikanaro**
- Better accounting for asymmetrical moon phases.
 - Creates enums and cleans up / replaces existing checks for capstone phase states.
 - Replace all existing uses of phase where cycle should have been checked intead.

**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Refactors core components of Moon Phase to better properly account for the [asymmetrical nature of moon cycles](https://www.bg-wiki.com/ffxi/Category:Moon_Phase).
- Adds a new getVanadielMoonCycle() method based on[ brief discussions](https://github.com/LandSandBoat/server/issues/8581) with Xaver and Sruon that translates to a new moon cycle out of the 12 possible cycle states. 
- Added a new ENUM for moon_cycles for use within the code base. 
- Combining the above two made IsMoonNew() and IsMoonFull() redundant so I removed them and replaced them in the code with a comparison outside of Core.
- Replaced all relevant calls to VanadielMoonPhase() with VanadielMoonCycle().
  -  Several places continue to use Phase when the actual % mattered and not the cycle state. Such as warhorse footprint and ~~chocobo digging~~. 

## Steps to test these changes

Summon Fenrir and use a BP Ward. Any of the first four wards should now properly account for asymmetrical edge cases. 
Go to Kuftal Tunnel and see the rock properly retreat using the new cycle function. 
Go to Konschat Highlands and see the behavior of Haty/Vran during Full/New Moons.
Go to Uleg Range and spawn White/Black Coney in their respective timeframes. 

### TODO: Impacts untested at this time:
 - Plenilune Embrace (Do BLU spells work? I can't even set it?)
 - ~~Chocobo Digging~~ _No longer relevant. Moved to Hobbies._

### CODE REVIEW

File: scripts/actions/abilities/pets/ecliptic_growl.lua [hunk 1]
```diff
@@ -14,19 +14,34 @@ abilityObject.onPetAbility = function(target, pet, petskill, summoner, action)
...
+    local moon = VanadielMoonCycle()
     local buffvalue = 1
-    if moon > 90 then
+    if moon == xi.moon_cycle.FULL_MOON then
         buffvalue = 7
-    elseif moon > 75 then
+    elseif
+        moon == xi.moon_cycle.GREATER_WANING_GIBBOUS or
+        moon == xi.moon_cycle.GREATER_WAXING_GIBBOUS
+    then
```
Comment (CriticalXI) [inline scripts/actions/abilities/pets/ecliptic_growl.lua]: I know you're just updating this existing block, but we could improve it to eliminate these long if-else chains. For example:

```lua
local moonBuffValues = 
{
    [xi.moon_cycle.FULL_MOON              ] = 7,
    [xi.moon_cycle.GREATER_WANING_GIBBOUS ] = 6,
    [xi.moon_cycle.LESSER_WANING_GIBBOUS  ] = 5,
    [xi.moon_cycle.FIRST_QUARTER          ] = 4,
    [xi.moon_cycle.THIRD_QUARTER          ] = 4,
    [xi.moon_cycle.GREATER_WANING_CRESCENT] = 3,
    [xi.moon_cycle.LESSER_WANING_CRESCENT ] = 2,
}

local buffvalue = moonBuffValues[VanadielMoonCycle()] or 1
```

The same approach could be applied to the other files that are using a similar logic.

File: scripts/enum/moon_cycle.lua [hunk 2]
```diff
@@ -0,0 +1,21 @@
+-----------------------------------
+-- Moon Cycle
+-----------------------------------
+xi = xi or {}
+
+---@enum xi.moon_cycle
+xi.moon_cycle =
```
Comment (Xaver-DaRed) [inline scripts/enum/moon_cycle.lua]: `xi.moonCycle`

That would be the standard

File: src/map/lua/luautils.cpp [hunk 3]
```diff
@@ -1649,6 +1648,90 @@ namespace luautils
...
+            }
+            if (phase > 5)
+            {
+                return 1; // Lesser Waxing Crescent
+            }
+            return 0; // New Moon
+        }
+
+        return 7; // Greater Waxing Gibbous is the least offensive as base case.
+    }
```
Comment (Xaver-DaRed) [inline src/map/lua/luautils.cpp:2072]: Does this even need to live in core?

You could just define a nice table in lua and create a very simple function to fetch the cycle.
Example
```lua
local moonTable =
{
    [0] =
    {
        { 50, 0 },
        {  0, 0 },
    },
    [1] =
    {
        { 94,  6 },
        { 77,  7 },
        { 61,  8 },
        { 41,  9 },
        { 25, 10 },
        { 11, 11 },
        {  0,  0 },
    },
    [2] =
    {
        { 90,  6 },
        { 72,  5 },
        { 56,  4 },
        { 39,  3 },
        { 22,  2 },
        {  6,  1 },
        {  0,  0 },
    },
}
```
And then to fetch the data
Something like
```lua
local direction
local phase

local table = moonTable[direction]
for i = 1, #table do
    if phase >= table[i][1] then
        return table[i][2]
    end
end
```
Food for thought

Comment (Rosienaikanaro) [reply src/map/lua/luautils.cpp:2072]: I agree with this and would actually prefer to remove it from core but it just didn't occur to me before. 

Where would the best place to build it out be? Somewhere like scripts/utils/common.lua under getVanaMidnight(day)?

**Review by Xaver-DaRed — APPROVED** [review]

**Comment by Xaver-DaRed** [comment]
This needs a rebase to resolve conflicts.
